Project/TaskList/app/src/main/java/Adapter/CalendarAdapter.java

```diff
@@ -0,0 +1,65 @@
+package Adapter;
+
+import android.graphics.Color;
+import android.os.Build;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.tasklist.CalendarUtils;
+import com.example.tasklist.R;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class CalendarAdapter extends RecyclerView.Adapter<CalendarViewHolder> {
+
+    private final ArrayList<LocalDate> days;
+    private OnItemListener mListener;
+
+    public CalendarAdapter(ArrayList<LocalDate> days, OnItemListener onItemListener) {
+        this.days = days;
+        mListener = onItemListener;
+    }
+    @NonNull
+    @Override
+    public CalendarViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.item, parent, false);
+        ViewGroup.LayoutParams layoutParams = view.getLayoutParams();
+        if (days.size() > 15) {
+            layoutParams.height = (int) (parent.getHeight() * 0.166666666);
```

Comment:
Is there a reason the amount of days modifies the height?

Reply:
No it doesn't have to be, it can be any number between 7 and 42, since there are 42 blocks in one page. Also for the weekly calendar page there's only seven blocks, so I randomly set up a number (also as what I learned from this calendar view tutorial